index.js

```diff
@@ -0,0 +1,175 @@
+let express = require('express');
+let cors = require('cors');
+let bodyParser = require('body-parser');
+let app = express();
+
+// This is necessary middleware to parse JSON into the incoming request body for POST requests
+app.use(bodyParser.json());
+
+/**
+ * Security Considerations:
+ * - CDS Services must implement CORS in order to be called from a web browser:
+ * - See details here: http://cds-hooks.org/#security
+ */
+app.use((request, response, next) => {
+  response.setHeader('Access-Control-Allow-Origin', '*');
+  response.setHeader('Access-Control-Allow-Methods', 'GET, POST, OPTIONS');
+  response.setHeader('Access-Control-Allow-Headers', 'X-Requested-With, Content-Type, Authorization');
+
+  // Pass to next layer of middleware
+  next();
+});
+
+/**
+ * Discovery Endpoint:
+ * - A GET request to the discovery endpoint, or URL path ending in '/cds-services'
+ * - This function should respond with definitions of each CDS Service for this app in JSON format
+ * - See details here: http://cds-hooks.org/#discovery
+ */
+app.get('/cds-services', (request, response) => {
+
+  // Example service to invoke the patient-view hook
+  const patientViewExample = {
+    hook: 'patient-view',
+    id: 'patient-view-example',
+    title: 'Example patient-view CDS Service',
+    description: 'Displays the name and gender of the patient',
+    prefetch: {
+      // Request the Patient FHIR resource for the patient in context, where the EHR fills out the prefetch template
+      // See details here: http://cds-hooks.org/#a-performance-tweak
+      requestedPatient: 'Patient/{{Patient.id}}'
+    }
+  };
+
+  // Example service to invoke the medication-prescribe hook
+  const medicationPrescribeExample = {
+    hook: 'medication-prescribe',
+    id: 'medication-prescribe-example',
+    title: 'Example medication-prescribe CDS Service',
+    description: 'Suggests prescribing Aspirin 81 MG Oral Tablets',
+  };
+
+  const discoveryEndpointServices = {
+    services: [ patientViewExample, medicationPrescribeExample ]
+  };
+  response.json(discoveryEndpointServices);
+});
+
+/**
+ * Patient View Example Service:
+ * - Handles POST requests to our patient-view-example endpoint
+ * - This function should respond with an array of card(s) in JSON format for the patient-view hook
+ *
+ * - Service purpose: Display a patient's first and last name, with a link to the CDS Hooks web page
+ */
+app.post('/cds-services/patient-view-example', (request, response) => {
```

Comment:
do we want to treat 'patient-view-example' as an id something like this?
`app.post('/cds-services/:serviceId', (request, response) => {`
The implications would be combing medication prescribe and patient view into a single route that then switches to different handlers.
I think it's more 'right' but maybe not simpler. So I'm on the fence.

Reply:
I considered this as well, but didn't think the "style" benefit outweighed the simplicity we want to go for.

Reply:
I would like to have a route for each hook rather than have a given route handle multiple hooks.